cmake/DetectOptions.cmake

```diff
@@ -177,22 +177,30 @@ endif()
 # Kokkos
 if(ADIOS2_USE_Kokkos)
   if(ADIOS2_USE_Kokkos STREQUAL AUTO)
-    find_package(Kokkos 3.7...<4.0 QUIET)
+    find_package(Kokkos QUIET)
   else()
-    find_package(Kokkos 3.7...<4.0 REQUIRED)
+    find_package(Kokkos REQUIRED)
   endif()
   if(Kokkos_FOUND)
-    set(ADIOS2_HAVE_Kokkos TRUE)
-    if(Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_HIP OR Kokkos_ENABLE_SYCL)
-      if(Kokkos_ENABLE_CUDA)
-        set(ADIOS2_HAVE_Kokkos_CUDA TRUE)
-        enable_language(CUDA)
+    # There is no easy way in CMake find_package with a range of multiple minor versions.
+    if (Kokkos_VERSION VERSION_LESS 3.7)
+      if(ADIOS2_USE_Kokkos STREQUAL ON)
```

Comment:
```suggestion
      if(ADIOS2_USE_Kokkos)
```

Reply:
I want to discriminate here against the AUTO value too.